src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/reroute/actions/UpdateFlowStatusAction.java

```diff
@@ -49,12 +50,25 @@ protected void perform(State from, State to, Event event, FlowRerouteContext con
             FlowStatus flowStatus = flow.computeFlowStatus();
             if (flowStatus != flow.getStatus()) {
                 dashboardLogger.onFlowStatusUpdate(flowId, flowStatus);
-                flowRepository.updateStatus(flowId, flowStatus);
+                flowRepository.updateStatus(flowId, flowStatus, getFlowStatusInfo(flowStatus, stateMachine));
+            } else if (FlowStatus.DEGRADED.equals(flowStatus)) {
```

Comment:
Why do we need to update statusInfo in cases when there where no actual status update? Why we calculate statusInfo separately from status itself? I.e. the code wich set some status to the flow, know why it has decided to set exactly this status... so it should set statusInfo field too.

Reply:
In the current implementation, the state DEGRADED can only be if the protected path was not found. This status (DEGRADED) can be set earlier. And because of this, we will not update statusInfo.